File: backend/internal/models/rental.go
package models

import "time"

type RentalStatus string

const (
	RentalStatusLocked    RentalStatus = "locked"
	RentalStatusActive    RentalStatus = "active"
	RentalStatusExpired   RentalStatus = "expired"
	RentalStatusCancelled RentalStatus = "cancelled"
)

type Rental struct {
	ID            string `json:"id"`
	StorageID     string `json:"storage_id"`
	StorageNumber string `json:"storage_number"`

	UserID    string `json:"user_id"`
	UserName  string `json:"user_name"`
	UserPhone string `json:"user_phone"`

	JKID string `json:"jk_id"`

	Months        int   `json:"months"`
	PricePerMonth int64 `json:"price_per_month"`
	TotalPaid     int64 `json:"total_paid"`

	StartsAt  time.Time    `json:"starts_at"`
	EndsAt    time.Time    `json:"ends_at"`
	Status    RentalStatus `json:"status"`
	CreatedAt time.Time    `json:"created_at"`
}

File: frontend/src/app/core/models/rental.ts
export type RentalStatus = 'locked' | 'active' | 'expired' | 'cancelled';

export interface Rental {
  id: string;

  storage_id: string;
  storage_number: string;

  user_id: string;
  user_name: string;
  user_phone: string;

  jk_id: string;

  months: number;
  price_per_month: number;
  total_paid: number;

  starts_at: string;
  ends_at: string;

  status: RentalStatus;
  created_at: string;
}
